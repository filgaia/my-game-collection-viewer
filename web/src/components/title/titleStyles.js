const titleStyles = (theme) => ({
    heroUnit: {
        backgroundColor: theme.palette.background.paper,
    },
    heroContent: {
        maxWidth: 600,
        margin: '0 auto',
        padding: `${theme.spacing.unit * 4}px 0 ${theme.spacing.unit * 3}px`,
    }
});

export default titleStyles;
