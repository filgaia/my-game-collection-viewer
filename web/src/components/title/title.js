// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// @material
import { Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
// @styles
import titleStyles from './titleStyles';

class Title extends Component {
    render() {
        const { classes } = this.props;

        return (
            <div className={classes.heroUnit}>
                <div className={classes.heroContent}>
                    <Typography component="h1" variant="h2" align="center" color="textPrimary" gutterBottom>
                        My Game Collection Viewer
                    </Typography>
                    <Typography variant="h6" align="center" color="textSecondary" paragraph>
                        Viewer for the great app My Game Collection!
                    </Typography>
                </div>
            </div>
        );
    }
}

Title.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(titleStyles)(Title);
