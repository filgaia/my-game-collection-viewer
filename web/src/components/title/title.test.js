import React from 'react';
import renderer from 'react-test-renderer';
import { Title } from './title';

describe('<Title />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Title
        classes={{}}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
