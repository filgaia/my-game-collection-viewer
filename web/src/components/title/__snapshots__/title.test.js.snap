// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> renders correctly 1`] = `
<div>
  <div>
    <h1
      className="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
    >
      My Game Collection Viewer
    </h1>
    <p
      className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter MuiTypography-paragraph css-1wu2fk9-MuiTypography-root"
    >
      Viewer for the great app My Game Collection!
    </p>
  </div>
</div>
`;
